From 98fd932a3210b6439cf8f1361ffe779b2731f927 Mon Sep 17 00:00:00 2001
From: Aaron Fabbri <fabbri@apache.org>
Date: Fri, 10 Mar 2017 16:00:42 -0800
Subject: [PATCH 2343/2848] HADOOP-14168 S3GuardTool tests should not run if
 S3Guard is not set up.  Contributed by Sean
 Mackrory

(cherry picked from commit 090924ae486a5877f4b0db7f0548e46d998e5404)

Change-Id: I635acbf43fa73822a65b274e937ad5fb65ca9533
---
 .../hadoop/fs/s3a/ITestS3ACredentialsInURL.java    |    6 +++---
 .../org/apache/hadoop/fs/s3a/S3ATestUtils.java     |   19 +++++++++++++------
 .../fs/s3a/s3guard/ITestS3GuardToolDynamoDB.java   |    5 -----
 .../hadoop/fs/s3a/s3guard/S3GuardToolTestBase.java |    2 +-
 4 files changed, 17 insertions(+), 15 deletions(-)

diff --git a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3ACredentialsInURL.java b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3ACredentialsInURL.java
index 3669dde..95d44cc 100644
--- a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3ACredentialsInURL.java
+++ b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/ITestS3ACredentialsInURL.java
@@ -38,7 +38,7 @@
 import java.nio.file.AccessDeniedException;
 
 import static org.apache.hadoop.fs.s3a.S3ATestConstants.TEST_FS_S3A_NAME;
-import static org.apache.hadoop.fs.s3a.S3ATestUtils.assumeS3GuardNotEnabled;
+import static org.apache.hadoop.fs.s3a.S3ATestUtils.assumeS3GuardState;
 
 /**
  * Tests that credentials can go into the URL. This includes a valid
@@ -68,7 +68,7 @@ public void testInstantiateFromURL() throws Throwable {
 
     // Skip in the case of S3Guard with DynamoDB because it cannot get
     // credentials for its own use if they're only in S3 URLs
-    assumeS3GuardNotEnabled(conf);
+    assumeS3GuardState(false, conf);
 
     String accessKey = conf.get(Constants.ACCESS_KEY);
     String secretKey = conf.get(Constants.SECRET_KEY);
@@ -127,7 +127,7 @@ public void testInvalidCredentialsFail() throws Throwable {
     Configuration conf = new Configuration();
     String fsname = conf.getTrimmed(TEST_FS_S3A_NAME, "");
     Assume.assumeNotNull(fsname);
-    assumeS3GuardNotEnabled(conf);
+    assumeS3GuardState(false, conf);
     URI original = new URI(fsname);
     URI testURI = createUriWithEmbeddedSecrets(original, "user", "//");
 
diff --git a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/S3ATestUtils.java b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/S3ATestUtils.java
index a8a5131..5a38541 100644
--- a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/S3ATestUtils.java
+++ b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/S3ATestUtils.java
@@ -24,6 +24,7 @@
 import org.apache.hadoop.fs.FileStatus;
 import org.apache.hadoop.fs.Path;
 import org.apache.hadoop.fs.permission.FsPermission;
+import org.hamcrest.core.Is;
 import org.junit.Assert;
 import org.junit.Assume;
 import org.junit.internal.AssumptionViolatedException;
@@ -374,17 +375,23 @@ public static Path createTestPath(Path defVal) {
   /**
    * Assume testing S3Guard is not enabled.
    */
-  public static void assumeS3GuardNotEnabled(Configuration originalConf)
-      throws URISyntaxException {
-    Assume.assumeFalse(getTestPropertyBool(originalConf, TEST_S3GUARD_ENABLED,
-        originalConf.getBoolean(TEST_S3GUARD_ENABLED, false)));
+  public static void assumeS3GuardState(boolean shouldBeEnabled,
+      Configuration originalConf) throws URISyntaxException {
+    boolean isEnabled = getTestPropertyBool(originalConf, TEST_S3GUARD_ENABLED,
+        originalConf.getBoolean(TEST_S3GUARD_ENABLED, false));
+    Assume.assumeThat("Unexpected S3Guard test state: shouldBeEnabled=" +
+        shouldBeEnabled + " and isEnabled =" + isEnabled,
+        shouldBeEnabled, Is.is(isEnabled));
 
     final String fsname = originalConf.getTrimmed(TEST_FS_S3A_NAME);
     Assume.assumeNotNull(fsname);
     final String bucket = new URI(fsname).getHost();
     final Configuration conf = propagateBucketOptions(originalConf, bucket);
-    Assume.assumeTrue(S3GUARD_METASTORE_NULL.equals(
-        conf.getTrimmed(S3_METADATA_STORE_IMPL, S3GUARD_METASTORE_NULL)));
+    boolean usingNullImpl = S3GUARD_METASTORE_NULL.equals(
+        conf.getTrimmed(S3_METADATA_STORE_IMPL, S3GUARD_METASTORE_NULL));
+    Assume.assumeThat("Unexpected S3Guard test state: shouldBeEnabled=" +
+        shouldBeEnabled + " but usingNullImpl=" + usingNullImpl,
+        shouldBeEnabled, Is.is(!usingNullImpl));
   }
 
   /**
diff --git a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/s3guard/ITestS3GuardToolDynamoDB.java b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/s3guard/ITestS3GuardToolDynamoDB.java
index 1356fe4..33cb001 100644
--- a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/s3guard/ITestS3GuardToolDynamoDB.java
+++ b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/s3guard/ITestS3GuardToolDynamoDB.java
@@ -30,11 +30,6 @@
 import java.util.Random;
 
 import static org.apache.hadoop.fs.s3a.s3guard.S3GuardTool.SUCCESS;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertFalse;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertTrue;
-import static org.junit.Assert.fail;
 
 /**
  * Test S3Guard related CLI commands against DynamoDB.
diff --git a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/s3guard/S3GuardToolTestBase.java b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/s3guard/S3GuardToolTestBase.java
index a465460..5254010 100644
--- a/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/s3guard/S3GuardToolTestBase.java
+++ b/hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/s3guard/S3GuardToolTestBase.java
@@ -66,7 +66,7 @@ protected S3AFileSystem getFs() {
   public void setUp() throws Exception {
     conf = new Configuration();
     fs = S3ATestUtils.createTestFileSystem(conf);
-
+    S3ATestUtils.assumeS3GuardState(true, getConf());
     ms = newMetadataStore();
     ms.initialize(fs);
   }
-- 
1.7.9.5

